File: classes/Calculate.php
<?php

namespace classes;

class Calculate
{
    /**
     * Method to get percentage to apply
     * @param int $numServices
     */
    public function getPercentage(int $numServices)
    {

        $percentageInit = 5;
        return ($numServices * $numServices) - $percentageInit;

    }

    /**
     * Method to calculate price of services
     * @param array $selection
     */
    public static function calculatePrice(array $selection)
    {


        $numServices = count($selection);
        $numLicences = array_sum($selection);
        $combinations = [];
        $combinations[$numServices] = intdiv($numLicences,$numServices);
        $combinations[$numLicences % $numServices] = 1;


        return $combinations;

    }


}

File: Python/classes/Calculate.py
import numpy as np


def getPercentage(numServices):
    percentageInit = 0.05
    return (percentageInit * numServices) - percentageInit if numServices < 5 else percentageInit * numServices


def calculatePrice(selection, priceLicense):
    total = 0
    numServices = len(selection)
    numLicences = np.sum(selection)
    combinations = []

    combinations[numServices] = (numLicences // numServices)
    combinations[numLicences % numServices] = 1

    for numLicense, repetitions in enumerate(combinations):
        total += repetitions * ((numLicense * priceLicense) * (1 - getPercentage(numLicense)))

    return total
